apps/aecore/src/aec_conductor.erl

```diff
@@ -417,8 +446,7 @@ preempt_if_new_top(#state{seen_top_block_hash = OldHash} = State, Publish) ->
             maybe_publish_top(Publish, NewHash),
             State1 = State#state{seen_top_block_hash = NewHash},
             State2 = kill_all_workers_with_tag(mining, State1),
-            State3 = kill_all_workers_with_tag(create_block_candidate, State2),
```

Comment:
How does `aec_block_generator` know that top changed when top changed because of block received from network? I could not find it. (I did not assume that distinct blocks necessarily have distinct set of txs, as that would seem a fragile assumption to me, so I did not explore that possibility for answering myself.)

Reply:
Since blocks received from another peer is still added to the chain through `aec_conductor` - `top_changed` means top changed due to block from network and `block_created` means the block was mined locally.